addons/l10n_it_edi/models/res_partner.py

```diff
@@ -135,8 +135,10 @@ def _l10n_it_edi_normalized_codice_fiscale(self, l10n_it_codice_fiscale=None):
         if l10n_it_codice_fiscale is None:
             self.ensure_one()
             l10n_it_codice_fiscale = self.l10n_it_codice_fiscale
-        if l10n_it_codice_fiscale and re.match(r'^IT[0-9]{11}$', l10n_it_codice_fiscale):
-            return l10n_it_codice_fiscale[2:13]
+        if l10n_it_codice_fiscale:
+            l10n_it_codice_fiscale = codicefiscale.compact(l10n_it_codice_fiscale)
+            if re.match(r'^IT[0-9]{11}$', l10n_it_codice_fiscale):
+                return l10n_it_codice_fiscale[2:13]
```

Comment:
```suggestion
        if l10n_it_codice_fiscale:
            if codicefiscale._code_re.match(l10n_it_codice_fiscale):
                # Personal codice
                return codicefiscale.compact(l10n_it_codice_fiscale)
            # Company codice
            return iva.compact(l10n_it_codice_fiscale)
```
I think this should cover all use cases.
Sorry for being picky, but I really would like to avoid using custom regex and slicing (which is often error prone), when there are existing (more robust) tools dedicated for that :nerd_face: